### PR TITLE
use full paths when calling exec command

### PR DESCRIPTION
**Pull request by rubys**
See:
  https://tip.golang.org/doc/go1.19#os-exec-path
  https://community.fly.io/t/unable-to-run-flyctl-launch/13504

### CODE REVIEW

**Comment by kzys** [comment]
Putting relative paths in PATH is bad. Go changed the behavior as you noted. Rust did it too in https://github.com/rust-lang/rust/pull/87704.

Is to coming from customers' own configuration (like ~/.bashrc) or tools (like Bundler)?

**Comment by rubys** [comment]
Putting relative paths in PATH is commonplace for Ruby.  Example: https://github.com/rbenv/rbenv/wiki/Understanding-binstubs#adding-project-specific-binstubs-to-path ; note the security concern mentioned on that page.

File: scanner/rails.go [hunk 1]
```diff
@@ -26,14 +28,39 @@ func configureRails(sourceDir string, config *ScannerConfig) (*SourceInfo, error
...
+
+		if err != nil {
+			return nil, errors.Wrap(err, "failure finding bundle executable")
+		}
+	}
+
+	ruby, err = exec.LookPath("ruby")
+	if err != nil {
+		if errors.Is(err, exec.ErrDot) {
+			bundle, err = filepath.Abs(ruby)
```
Comment (kzys) [inline scanner/rails.go:48]: bundle -> ruby

File: scanner/rails.go [hunk 2]
```diff
@@ -71,9 +98,21 @@ func configureRails(sourceDir string, config *ScannerConfig) (*SourceInfo, error
 			},
 		}
 	} else {
+		// find absolute path to rake executable
+		rake, err := exec.LookPath("rake")
+		if err != nil {
+			if errors.Is(err, exec.ErrDot) {
+				bundle, err = filepath.Abs(ruby)
```
Comment (kzys) [inline scanner/rails.go:105]: bundle -> rake?

**Review by kzys — CHANGES_REQUESTED** [review]

**Comment by kzys** [comment]
Oh rbenv...

**Review by kzys — APPROVED** [review]

**Pull request merged**